src/lib.rs

```diff
@@ -505,4 +514,60 @@ mod test {
         assert!(res.is_err(), "timed out");
         assert_eq!(*l.lock().await, vec!["child gonna nap"]);
     }
+
+    // This serves as a regression check for https://github.com/pchickey/task-group/issues/3
+    // I'm not sure how fragile this test will be, since it may
+    // depend on Rust and LLVM's ability to optimize out unused
+    // allocations like big_object.
```

Comment:
A few ideas if it's important to make `big_future` less fragile:
- You could use a function like [`criterion::black_box`](https://docs.rs/criterion/0.3.5/criterion/fn.black_box.html) on big_object after the await point to make sure it never gets optimized out. 
  - If you don't wanna add a dependency, you could print `big_object` after the await point. The print would not show up in test output unless the test failed.
- You could make a struct that holds some data and implements future manually